runtime/src/active_stats.rs

```diff
@@ -0,0 +1,64 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[derive(Debug, Default)]
+pub struct ActiveStats {
+    clean: AtomicUsize,
+    shrink: AtomicUsize,
+    hash: AtomicUsize,
+    flush: AtomicUsize,
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum ActiveStatItem {
+    Clean,
+    Shrink,
+    Hash,
+    Flush,
+}
+
+/// sole purpose is to handle 'drop' so that stat is decremented when self is dropped
+pub struct ActiveState<'a> {
+    stats: &'a ActiveStats,
+    item: ActiveStatItem,
+}
+
+impl<'a> Drop for ActiveState<'a> {
+    fn drop(&mut self) {
+        self.stats.log(self.item, false);
+    }
+}
+
+impl ActiveStats {
+    #[must_use]
+    /// create a stack object to set the state to increment stat initially and decrement on drop
+    pub fn get_state(&self, stat: ActiveStatItem) -> ActiveState<'_> {
```

Comment:
I like this idea a lot. This fn (and type) name confused me at first though, since the "state" that I'm getting is not really meant to be used at all. It follows the similar RAII patters, like Mutex, and Mutex obviously has `lock()`, which doesn't really apply here. But maybe the Guard type does?

I.e. the type could be `ActiveStatGuard`, and this fn could be `activate()` or something?